### PR TITLE
GS/HW: Skip draws on reset until scissor is set

### LINKED ISSUE #8602: [BUG]: God of War 2 dark chests
### Describe the Bug

When using Progressive scan mode in HW mode chests appear dark, missing shading or something. I would provide a state but the bug doesn't occur when loading one. I noticed when progressive scan is activated there is a brief flash of garbage. Maybe related to Radiata issue?

### Reproduction Steps

Turn on progressive scan mode and get to a chest in the game, or load the GS dump. 

### Expected Behavior

Chest should not be dark. 

### PCSX2 Revision

v1.7.4348

### Operating System

Windows 11

### If Linux - Specify Distro

_No response_

### CPU

i7 8086k

### GPU

Nvidia GTX 1080ti

### GS Settings

Default HW

### Emulation Settings

Default

### GS Window Screenshots

God of War II_SCUS-97481_20230407060553

### Logs & Dumps

[GoW 2 GS.zip](https://github.com/PCSX2/pcsx2/files/11181691/GoW.2.GS.zip)

### PR DESCRIPTION
**Pull request by stenzek**
### Description of Changes

~~My **guess** is that the drawing environment registers get left alone, and only the privileged registers are reset.~~

See the notes in the diff/source.

```
// Basically, GOW does a 32x448 draw after resetting the GS, thinking the PSM for the framebuffer is going to be set to C24,
// therefore the alpha bits get left alone. Because of the reset, in PCSX2, it ends up as C32, and the TC gets confused,
// leading to a later texture load using this render target instead of local memory. It's a problem because the game uploads
// texture data on startup to the beginning of VRAM, and never overwrites it.
//
// In the software renderer, if we let the draw happen, it gets scissored to 1x1 (because the scissor is inclusive of the
// upper bounds). This doesn't seem to destroy the chest texture, presumably it's further out in memory.
//
// Hardware test show that VRAM gets corrupted on CSR reset, but the first page remains intact. We're guessing this has something
// to do with DRAM refresh, and perhaps the internal counters used for refresh also getting reset. We're obviously not going
// to emulate this, but to work around the aforementioned issue, in the hardware renderers, we set the scissor to an out of
// bounds value. This means that draws get skipped until the game sets a proper scissor up, which is definitely going to happen
// after reset (otherwise it'd only ever render 1x1).
```

### Rationale behind Changes

Closes #8602.

### Suggested Testing Steps

Test progressive switches in various games. It seems to also fix the garbage frame you might see.

~~However, this needs to be hardware tested, to confirm whether the environment resets or not. If it does, then we have a more serious issue...~~

### CODE REVIEW

**Comment by Goatman13** [comment]
I don't know if this change anything because i barely understand GS, but there are at least 2 emulators that set all framebuffer mask bits to 1 for frame, and buffer mask bit to 1 for zbuf on reset. While pcsx2 set them to 0 if i understand correctly. Of course they can be wrong, so you need hardware test anyway, even if that work.

**Comment by refractionpcsx2** [comment]
I'm not sure if setting all the bits to 1 makes much sense, but we can test it for sure.  Fobes (one of our guys) is doing some tests currently, so we're going to check how much needs to be set in order to draw *something*

in theory we should be able to try and draw 1 page at 0x0000 without touching the frame, and if it doesn't draw anything, then the mask is set, otherwise it's not.

**Comment by stenzek** [comment]
Setting FBMSK+ZMSK would indeed fix the issue with GOW. However, it's still inconclusive as to whether the hardware behaves this way.

Basically, GOW does a 32x448 draw after resetting the GS, thinking the PSM for the framebuffer is going to be set to C24, therefore the alpha bits get left alone. Because of the reset, in PCSX2, it ends up as C32, and the TC gets confused, leading to a later texture load using this render target instead of local memory. It's a problem because the game uploads texture data on startup to the beginning of VRAM, and never overwrites it.

In the software renderer, if we let the draw happen, it gets scissored to 1x1 (because the scissor is inclusive of the upper bounds). This doesn't seem to destroy the chest texture, presumably it's further out in memory.

So, tldr, it's possible that the console *does* reset everything to zero, because doing so doesn't break this case. It's just a pain for our TC.

**Comment by stenzek** [comment]
OP updated, we found a less crappy workaround. Dump runner says it's fine, just needs some quick manual testing.

**Comment by refractionpcsx2** [comment]
Just for luls btw, we put a reset in a loop, to see how much the VRAM refresh thing would affect it... welll, this was a filled red square xD

image

**Review by JordanTheToaster — APPROVED** [review]

**Comment by Goatman13** [comment]
> Also @Goatman13 the masks don't get set, you can draw right after a reset :)

Not the first time when official $ony emus are busted. Good work on figuring it out. :)
